File: frontend/src/services/settingsService.js
import apiClient from './api';

export default {
  // Get all settings
  getSettings() {
    return apiClient.get('/settings');
  },

  // Update settings
  updateSettings(settings) {
    return apiClient.put('/settings', settings);
  },

  // Update API configuration
  updateApiConfig(index, config) {
    return apiClient.put(`/settings/api/${index}`, config);
  },

  // Update font settings
  updateFontSettings(fontSettings) {
    return apiClient.put('/settings/fonts', fontSettings);
  },

  // Update import settings
  updateImportSettings(importSettings) {
    return apiClient.put('/settings/import', importSettings);
  },

  // Reset settings to default
  resetSettings() {
    return apiClient.post('/settings/reset');
  },

  // Backup all data
  backupData() {
    return apiClient.post('/settings/backup');
  },

  // Restore data from backup
  restoreData(filename) {
    return apiClient.post('/settings/restore', { filename });
  },

  // Update data storage path
  updateDataPath(path) {
    return apiClient.put('/settings/data-path', { path });
  }
};

File: backend-springboot/src/main/java/com/quizgen/app/setting/service/SettingsService.java
package com.quizgen.app.setting.service;

import com.quizgen.app.setting.dto.UpdateApiConfigRequest;

import java.util.List;
import java.util.Map;

public interface SettingsService {

    Map<String, Object> getSettings();

    Map<String, Object> updateSettings(Map<String, Object> request);

    Map<String, Object> updateApiConfig(int index, UpdateApiConfigRequest request);

    Map<String, Object> updateFontSettings(Map<String, Object> request);

    Map<String, Object> updateImportSettings(Map<String, Object> request);

    Map<String, Object> resetSettings();

    Map<String, Object> backupData();

    Map<String, Object> getBackupJobStatus(String jobNo);

    Map<String, Object> restoreData(String filename);

    Map<String, Object> updateDataPath(String path);

    List<Map<String, Object>> getBackupFiles();
}
